ui/app/components/identity/popup-policy.js

```diff
@@ -3,32 +3,47 @@
  * SPDX-License-Identifier: BUSL-1.1
  */
 
-import { alias } from '@ember/object/computed';
-import { computed } from '@ember/object';
-import Base from './_popup-base';
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { service } from '@ember/service';
+import errorMessage from 'vault/utils/error-message';
+import { tracked } from '@glimmer/tracking';
 
-export default Base.extend({
-  model: alias('params.firstObject'),
-  policyName: computed('params', function () {
-    return this.params.objectAt(1);
-  }),
+export default class IdentityPopupPolicy extends Component {
+  @service flashMessages;
+  @tracked showConfirmModal = false;
 
-  messageArgs(model, policyName) {
-    return [model, policyName];
-  },
+  onSuccess(policyName, modelId) {
+    if (this.args.onSuccess) {
+      this.args.onSuccess();
+    }
+    this.flashMessages.success(`Successfully removed '${policyName}' policy from ${modelId} `);
+  }
+  onError(err, policyName) {
+    if (this.args.onError) {
+      this.args.onError();
+    }
+    const error = errorMessage(err);
+    this.flashMessages.danger(`There was a problem removing '${policyName}' policy - ${error}`);
+  }
 
-  successMessage(model, policyName) {
-    return `Successfully removed '${policyName}' policy from ${model.id} `;
-  },
+  transaction() {
```

Comment:
should this be async as you call an await on it below?

Reply:
Since it's returning a promise (`model.save`) and we're not using `await` within this method there's no need for `async` here